src/commands/scaling/addScaleRule/AddScaleRuleStep.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ScaleRule, Template } from "@azure/arm-appcontainers";
+import { AzureWizardExecuteStep, nonNullProp, parseError } from "@microsoft/vscode-azext-utils";
+import { Progress, window } from "vscode";
+import { ScaleRuleTypes } from "../../../constants";
+import { ext } from "../../../extensionVariables";
+import { localize } from "../../../utils/localize";
+import { updateContainerApp } from "../../updateContainerApp";
+import { IAddScaleRuleWizardContext } from "./IAddScaleRuleWizardContext";
+
+export class AddScaleRuleStep extends AzureWizardExecuteStep<IAddScaleRuleWizardContext> {
+    public priority: number = 100;
+
+    public async execute(context: IAddScaleRuleWizardContext, _progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
+        const adding = localize('addingScaleRule', 'Adding "{0}" {1} type rule to "{2}"...', context.ruleName, context.ruleType, context.containerApp.name);
+        const added = localize('addedScaleRule', 'Successfully added "{0}" {1} type rule to "{2}".', context.ruleName, context.ruleType, context.containerApp.name);
+        const failed = localize('failedAddScaleRule', 'Failed to add "{0}" {1} type rule to "{2}".', context.ruleName, context.ruleType, context.containerApp.name);
+
+        const template: Template = context.containerApp.data.template || {};
+        template.scale = context.scale.data || {};
+        template.scale.rules = context.scaleRuleGroup.data || [];
+
+        const scaleRule: ScaleRule = this.buildRule(context);
+        this.integrateRule(context, template.scale.rules, scaleRule);
+
+        try {
+            ext.outputChannel.appendLog(adding);
+            await updateContainerApp(context, context.containerApp, { template });
+            void window.showInformationMessage(added);
+            ext.outputChannel.appendLog(added);
+            context.scaleRule = scaleRule;
+        } catch (error) {
```

Comment:
Oh I see, I should have checked how createChildImpl was called, it looks like it just swallows the error without display it which covers the issue I was trying to avoid by doing this (having two errors pop up because of it not being percolated up properly)